components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -269,6 +269,7 @@ public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumer
                     app.setAudiences(application.getAudiences());
                     app.setPkceMandatory(application.getPkceMandatory());
                     app.setPkceSupportPlain(application.getPkceSupportPlain());
+                    app.setAuthorizationCodeValidityPeriod(application.getAuthorizationCodeValidityPeriod());
```

Comment:
Try to follow the pattern used for other token expiry times.
ie. validate and set default if the sent value is not valid.